src/BtcProofUtils.sol

```diff
@@ -254,7 +254,6 @@ library BtcProofUtils {
             offset += 8;
             uint256 nOutScriptBytes;
             (nOutScriptBytes, offset) = readVarInt(rawTx, offset);
-            require(nOutScriptBytes <= 32, "Scripts over 32 bytes unsupported");
             txOut.scriptLen = uint32(nOutScriptBytes);
             txOut.script = bytes32(rawTx[offset:offset + nOutScriptBytes]);
```

Comment:
won't this conversion to `bytes32` fail if the slice is longer than 32 bytes?

we can still remove the restriction on input scripts, but we may have to do something like


```
if (nOutScriptBytes > 32) txOut.script = 0x0;
else txOu.script = bytes32(...);
```

... and just add a comment on the `script` field in the struct that it'll be 0 if `scriptLen` is over 32

Reply:
Yea wasn't sure how much surgery to do if tests were looking @ this

Is there a reason to parse in txes at all? We don't seem to verify anything from them

Reply:
I think the conversion truncates, so it doesn't break, its just not 100% correct in the struct. In any case, we validate for the only tx that matters in the function that calls it. Can clean this up before we go live